### ISSUE TITLE
Migrate all certification flags on users

### ISSUE THREAD

**Issue by raisedadead** [freeCodeCamp/freeCodeCamp#38609]
**Context:**

The current (pre-2020) list of certificate flags are:

```txt
  // Certifications - 2019
  isRespWebDesignCert
  isJsAlgoDataStructCert
  isFrontEndLibsCert
  isDataVisCert
  isApisMicroservicesCert
  isInfosecQaCert

  isFullStackCert // <-- Claimed only if you have 2019 certs

  // Certifications - 2018 (and prior)
  isFrontEndCert
  isBackEndCert
  is2018DataVisCert
  is2018FullStackCert // <-- Never claimed by any user.
```

**Impact**

This has some deep impact on the way we create and release certifications, and leading to avoidable bugs.

**Proposal**

We should add the certification flags with the year they are released by default. This will allow us to scale as and when needed.

The certs should be backwards compatible for everyone by **default**.

For example, A new python certification like 'Scientific Computing with Python' should have a flag like:

```txt
  is2020SciCompPyCert
```

and all our current certifications should become:

```diff
  // Certifications - 2019
- isRespWebDesignCert
+ is2019RespWebDesignCert
- isJsAlgoDataStructCert
+ is2019JsAlgoDataStructCert
- isFrontEndLibsCert
+ is2019FrontEndLibsCert
- isDataVisCert
+ is2019DataVisCert
- isApisMicroservicesCert
+ is2019ApisMicroservicesCert
- isInfosecQaCert
+ is2019InfosecQaCert

- isFullStackCert // <-- Claimed only if you have 2019 certs
+ is2019FullStackCert // <-- Claimed only if you have 2019 certs

  // Certifications - 2018 (and prior)
- isFrontEndCert
+ is2018FrontEndCert
- isBackEndCert
+ is2018BackEndCert
  is2018DataVisCert
  is2018FullStackCert // <-- Never claimed by any user.
```

and finally when we add the planned new certifications:

```txt
  is2020SciCompPyCert
  is2020DataAnalysisPyCert
  is2020MachineLearningPyCert
```
which allows us to split up the "InfoSec & QA" cert to "InfoSec" cert and "QA" cert individually:

```txt
- isInfosecQaCert
+ is2019InfosecQaCert


+ is2020InfosecCert
+ is2020QaCert
```

and so on.

Implementing this would mean doing a one-time migration on all user records.

**Comment by raisedadead** [comment]
Thanks, I wanted some initial feedback from the team - opening up for contributions and suggestions.

**Comment by raisedadead** [comment]
We are reconsidering to do v6, v7 like versioning on the certifications.
/cc @freeCodeCamp/dev-team

**Comment by scissorsneedfoodtoo** [comment]
@raisedadead, was just coming back to start changing the slugs in https://github.com/freeCodeCamp/freeCodeCamp/pull/39026 and saw that it's been merged. Would be happy to create a new PR to replace 2020___ with v7 in a fresh PR.

Should we also use the same v__ versioning system for the flags in the DB? Or stick with years there and use v7 just for the slugs?

**Comment by raisedadead** [comment]
Hi @scissorsneedfoodtoo 

Yes, I was about to get started on this, but if you are up for it, sure go ahead.

The idea is to use versioning but instead of the intitial thought of using year, because it could have unintended effects.

Here is what we are thinking:
 
v5 (legacy - pre 2018 certs)
v6 (2018 - 2019 certs)
v7 (2020 certs)

and so on. We also think we would be doing the same approach, i.e add the new version right away and unblock Python curriculuim, followed by a separate plan to update all the flags/db etc. later this year.

**Comment by moT01** [comment]
Do we still want to do this @mrugesh? Or would we be fine to just use that naming scheme moving forward so we can close this?

**Comment by ahmaxed** [comment]
We need a naming convention. Once that is finalized, this can go in with the database normalization sprint.

**Comment by ShaunSHamilton** [comment]
After discussing this, it might be better to convert our `is<cert>` to:

```json
certifications: [
  {
    id: <certification_id>,
    timestamp: 1234567890
  }
]
```

We do not need the human-readable names in the MongoDB database. Considering when we query this data, we might even be better putting this in its own collection.

The CDB will hold the human-readable names (e.g. title, dashed_name).
